code/handlers/trackservers.q

```diff
@@ -175,13 +173,16 @@ retrydiscovery:{
 // Called by the discovery service when it restarts
 autodiscovery:{if[DISCOVERYRETRY>0; .servers.retrydiscovery[]]}
 
+//check if .proc.getattributes exists in the nontorqprocess
+checknontorqattr:{(`proc in key`)&(`getattributes in key`.proc)};
+
 // Attempt to make a connection for specified row ids
 retryrows:{[rows]
     // opencon, amends global tables, cannot be used inside of a select statement
     handles:.servers.opencon each exec hpup from`.servers.SERVERS where i in rows;
-    update lastp:.proc.cp[],w:handles from`.servers.SERVERS where i in rows;
-        update attributes:{$[null x;()!();@[x;(`.proc.getattributes;`);()!()]]} each w,startp:?[null w;0Np;.proc.cp[]] from `.servers.SERVERS where i in rows;
-        if[ count connectedrows:select from `.servers.SERVERS where i in rows, .dotz.liveh0 w;
+    update lastp:.proc.cp[],w:handles from`.servers.SERVERS where i in rows; 
+    update attributes:{$[null x;()!();@[x;(checknontorqattr;())];@[x;(`.proc.getattributes;`);()!()];()!()]} each w,startp:?[null w;0Np;.proc.cp[]] from `.servers.SERVERS where i in rows;
```

Comment:
Haven't tested this to be sure, but I think this can be further simplified something like

```q
$[not[null x]&@[x;(checknontorqattr;`);0b];@[x;(`.proc.getattributes;`);()!()];()!()]
```

This lambda is possibly approaching the length where it could be split out into it's own function? Could be more nicely laid out that way to be more readable I think